### PR TITLE
Python 2 compatibility

### PR DESCRIPTION
**Pull request by eight04**

### CODE REVIEW

**Comment by eight04** [comment]
There is a problem that I think it is not fixable without breaking changes.

```
======================================================================
FAIL: test_is_png (__main__.Main) [file name]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 28, in test_is_png
    assert is_png(str(file))
AssertionError

----------------------------------------------------------------------
```

In Python 2, `bytes` is identical to `str`, so this would fail:

```python
if isinstance(file, bytes):
  png_head = file[:8]
else:
  with open(file, "rb") as f:
    png_head = f.read(8) 
```

@robmcmullen Any ideas?

**Comment by robmcmullen** [comment]
The only thing that comes to mind is to use try/except blocks, so instead of checking with isinstance, you just keep trying stuff based on the methods you need rather than the specific object types.

**Comment by eight04** [comment]
If we use try/except, whenever a `bytes` object is sent to this library, the library would make a request to the file system. It doesn't make sense IMHO.

Here is a breaking change solution:

1. Just always use `bytes` for input/output and leave file system/type conversion logic to users.
2. Provide some shortcuts to create APNG from files:

    ```python
    # read the file into bytes and convert it to PNG automatically.
    APNG.from_files(["1.jpg", "2.jpg", "3.jpg"], delay=100).save("result.png")
    
    # also accept a list of options
    APNG.from_files([
      {
        "filename": "1.jpg",
        "delay": 100
      },
      {
        "filename": "2.jpg",
        "delay": 200
      },
      {
        "filename": "3.jpg",
        "delay": 300
      }
    ])
    ```

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once).You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+disable=
```
Comment (eight04) [inline .pylintrc]: Remove unused settings.

File: apng/__init__.py [hunk 2]
```diff
@@ -345,54 +379,61 @@ def open(cls, file):
...
-			elif type == "fcTL":
-				if any(type == "IDAT" for type, data in frame_chunks):
+			elif type_ == "fcTL":
+				if any(type_ == "IDAT" for type_, data in frame_chunks):
 					# IDAT inside chunk, go to next frame
 					frame_chunks.append(end)
 					frames.append((PNG.from_chunks(frame_chunks), control))
 					
 					control = FrameControl.from_bytes(data[12:-4])
+					# pylint: disable=typecheck
```
Comment (eight04) [inline apng/__init__.py]: Link the issue here: https://github.com/PyCQA/pylint/issues/2072

File: README.rst [hunk 3]
```diff
@@ -1,36 +1,40 @@
...
+- `Pillow <https://github.com/python-pillow/Pillow>`__ - **Optional**. You can still use pyAPNG without PIL but it can only read PNG files.
 
 Development dependencies
 ------------------------
 
--  `pngcheck <http://www.libpng.org/pub/png/apps/pngcheck.html>`_
--  See requirements.txt for other dev-dependencies.
+- `pngcheck <http://www.libpng.org/pub/png/apps/pngcheck.html>`_
+- See requirements.txt for other dev-dependencies.
+- See requirements-py2.txt for dev-dependencies in Python 2 environment.
```
Comment (eight04) [inline README.rst]: We now only use one file.

File: docs/conf.py [hunk 4]
```diff
@@ -8,6 +8,6 @@
 add_module_names = False
 extensions = ["sphinx.ext.intersphinx", "sphinx.ext.autodoc"]
 master_doc = "index"
-autodoc_member_order = "bysource"
+# autodoc_member_order = "bysource"
```
Comment (eight04) [inline docs/conf.py]: Remove this line entirely.

File: test.py [hunk 5]
```diff
@@ -45,36 +37,36 @@ def iter_frames(dir):
...
-			with self.subTest(dir.stem):
-				im = APNG.open(dir.joinpath("animated.png"))
-				property = dir.joinpath("property.json")
-				if property.exists():
-					property = json.loads(property.read_text())
-					for key, value in property.items():
+		for dir_ in pathlib.Path("test").iterdir():
+			with self.subTest(dir_.stem):
+				im = APNG.open(dir_.joinpath("animated.png"))
+				options_path = dir_.joinpath("property.json")
```
Comment (eight04) [inline test.py]: `options_file` is better.

**Pull request merged**